parsing-utilities/src/main/java/gov/nasa/jpl/aerie/json/JsonObjectParser.java

```diff
@@ -0,0 +1,34 @@
+package gov.nasa.jpl.aerie.json;
+
+import javax.json.JsonObject;
+import javax.json.JsonValue;
+import java.util.Objects;
+
+public interface JsonObjectParser<T> extends JsonParser<T> {
+  @Override
+  JsonObject unparse(final T value);
+
+  @Override
+  default <S> JsonObjectParser<S> map(final Iso<T, S> transform) {
+    Objects.requireNonNull(transform);
+
+    final var self = this;
```

Comment:
(It's a common pattern [in JavaScript, too](https://stackoverflow.com/a/338106), which is where I picked it up. It's less necessary these days in JS because of the prevalence of arrow functions.)